source/retryable-writes/retryable-writes.rst

```diff
@@ -137,21 +137,20 @@ check the user’s command document to determine if it is a write, nor check
 whether the server is new enough to support a transaction ID for the command.
 The method should simply send the user’s command document to the server as-is.
 
-This specification does not affect write commands executed outside of a
-DriverSession or within a DriverSession where retryable writes have not been
-enabled.
+This specification does not affect write commands executed in an application
+where retryable writes have not been enabled.
```

Comment:
> If your driver offers a generic command method on your database object, it MUST NOT automatically add a transaction ID. The generic command method MUST NOT check the user’s command document to determine if it is a write, nor check whether the server is new enough to support a transaction ID for the command. The method should simply send the user’s command document to the server as-is.

IIRC, this paragraph was added during the draft stage based on @ajdavis' feedback.

In this section, I'm alluding to read and write commands as indicated in the Read and Write Concern spec (a read command is one that solicits a `readConcern` parameter). To that end, there are commands that are neither read nor write, such as `ping` and `isMaster`. A generic command helper is the only driver facility for sending a raw command to the server, and therefore should not concern itself with injecting options into the command document.

From the context of libmongoc, there are methods for executing arbitrary read, write, and read/write commands. These function like `runCommand()` but _do_ inject read- and write-specific options such as `readConcern`.

I'd argue that the Driver Session specification is free to mandate that `runCommand()` always inject a session ID, but we don't need to mandate injection of a transaction ID (i.e. the combination of session ID and transaction number) in this spec.